form.js

```diff
@@ -0,0 +1,296 @@
+	InputIsCorrect: false,
+};
+
+const ComparingPasswordsSyntax = {
+	valueInput: document.querySelector("#password2"),
+	errorName: document.querySelector(".error-password2"),
+	iconCheck: document.querySelector(".icon-password2-correct"),
+	iconX: document.querySelector(".icon-password2-incorrect"),
+	InputIsCorrect: false,
+};
+
+const citySyntax = {
+	valueInput: document.querySelector("#city"),
+	errorName: document.querySelector(".error-city"),
+	iconCheck: document.querySelector(".icon-city-correct"),
+	iconX: document.querySelector(".icon-city-incorrect"),
+	InputIsCorrect: false,
+};
+
+const emailSyntax = {
+	valueInput: document.querySelector("#email"),
+	errorName: document.querySelector(".error-email"),
+	iconCheck: document.querySelector(".icon-email-correct"),
+	iconX: document.querySelector(".icon-email-incorrect"),
+	InputIsCorrect: false,
+};
+
+const allError = [
+	nameSyntax.errorName,
+	passwordSyntax.errorName,
+	ComparingPasswordsSyntax.errorName,
+	citySyntax.errorName,
+	emailSyntax.errorName,
+];
+
+const everyGoodPath = syntax => {
+	syntax.valueInput.classList.remove("underline");
+	syntax.errorName.style.visibility = "hidden";
+	syntax.errorName.textContent = "";
+	syntax.iconCheck.style.visibility = "visible";
+	syntax.iconX.style.visibility = "hidden";
+	syntax.InputIsCorrect = true;
+	allInputIsCorrect();
+};
+const everyError = (syntax, error) => {
+	syntax.valueInput.classList.add("underline");
+	syntax.errorName.style.visibility = "visible";
+	syntax.errorName.textContent = error;
+	syntax.iconCheck.style.visibility = "hidden";
+	syntax.iconX.style.visibility = "visible";
+	syntax.InputIsCorrect = false;
+};
+
+const sendGoodPath = whatKindOfInput => {
+	switch (whatKindOfInput) {
+		case "name":
+			everyGoodPath(nameSyntax);
+			break;
+
+		case "password":
+			everyGoodPath(passwordSyntax);
+			break;
+
+		case "ComparingPasswords":
+			everyGoodPath(ComparingPasswordsSyntax);
+			break;
+
+		case "city":
+			everyGoodPath(citySyntax);
+			break;
+
+		case "email":
+			everyGoodPath(emailSyntax);
+			break;
+	}
+};
+
+const sendError = (whatKindOfInput, error) => {
+	switch (whatKindOfInput) {
+		case "name":
+			everyError(nameSyntax, error);
+			break;
+
+		case "password":
+			everyError(passwordSyntax, error);
+			break;
+
+		case "ComparingPasswords":
+			everyError(ComparingPasswordsSyntax, error);
+			break;
+
+		case "city":
+			everyError(citySyntax, error);
+			break;
+
+		case "email":
+			everyError(emailSyntax, error);
+			break;
+	}
+};
+
+const checkUserName = () => {
+	if (!RegEx.CHECKING_SPECIAL_CHARACTER.test(nameSyntax.valueInput.value)) {
+		if (
+			nameSyntax.valueInput.value.length >
+			theMinimumNumberOfCharactersInTheNickname
+		) {
+			sendGoodPath("name");
+		} else {
+			sendError(
+				"name",
+				everyErrorText.THE_NAME_MUST_CONTAIN_AT_LEAST_3_LETTERS
+			);
+		}
+	} else {
+		sendError("name", everyErrorText.FORBIDDEN_SPECIAL_CHARACTERS);
```

Comment:
I believe "name" should be exported to an constant in a case you would like you change it in the future so you can change all at once instead of replacing it in 3 or more places